src/contracts/core/StakeRootCompendium.sol

```diff
@@ -0,0 +1,134 @@
+            require(delegation.isOperator(operators[i]), "AVSSyncTree.getOperatorSetRoot: operator not registered");
+            require(avsDirectory.avsOperatorStatus(avs,operators[i]) == IAVSDirectory.OperatorAVSRegistrationStatus.REGISTERED, "AVSSyncTree.getOperatorSetRoot: operator not registered to AVS");
+            require(avsDirectory.isMember(avs,operators[i],operatorSetId), "AVSSyncTree.getOperatorSetRoot: operator not in operator set");
+
+            // shares associated with this operator and these strategies
+            uint256 weightedStrategyShareSum = _calculateWeightedStrategyShareSum(operators[i], strategies, getMultipliers(operatorSetId, strategies));
+            bytes32 operatorRoot = keccak256(abi.encodePacked(operators[i], weightedStrategyShareSum));
+            operatorLeaves[i] = keccak256(abi.encodePacked(operators[i], operatorRoot));     
+        }
+        return Merkle.merkleizeKeccak256(operatorLeaves);
+    }
+
+    function verifySnarkProof(
+        bytes calldata _journal,
+        bytes calldata _seal
+    ) external {
+        IRiscZeroVerifier(verifier).verify(
+                _seal,
+                imageId,
+                sha256(_journal)
+            );
+        
+        emit SnarkProofVerified(_journal, _seal);
+    }
+
+    //TODO: getting an error here:
+    // Error: Unimplemented feature (/solidity/libsolidity/codegen/ArrayUtils.cpp:240):
+    // Copying of type struct IStakeRootCompendium.StrategyAndMultiplier calldata[] calldata to storage not yet supported.
+    function setStrategiesAndMultipliers(
+        uint32 operatorSetId,
+        StrategyAndMultiplier[] calldata strategiesAndMultipliers
+    ) external isOperatorSetAVS(msg.sender) {
+        for (uint256 i = 0; i < strategiesAndMultipliers.length; i++) {
+            operatorSetIdToStrategyToMultiplier[operatorSetId][strategiesAndMultipliers[i].strategy] = strategiesAndMultipliers[i].multiplier;
+        }
+    }
+
+
+    function getMultipliers(
+        uint32 operatorSetId,
+        IStrategy[] calldata strategies
+    ) public view returns (uint96[] memory) {
+        uint96[] memory multipliers = new uint96[](strategies.length);
+        for (uint256 i = 0; i < strategies.length; i++) {
+            multipliers[i] = operatorSetIdToStrategyToMultiplier[operatorSetId][strategies[i]];
+        }
+        return multipliers;
+    }
+
+    function setVerifier(address _verifier) external onlyOwner {
+        address oldVerifier = verifier; 
+        verifier = _verifier;
+        emit VerifierChanged(oldVerifier, verifier);
+    }
+
+    function setImageId(bytes32 _imageId) external onlyOwner {
+        bytes32 oldImageId = imageId;
+        imageId = _imageId;
+        emit ImageIdChanged(oldImageId, imageId);
+    }
+
+    function _calculateWeightedStrategyShareSum(address operator, IStrategy[] memory strategies, uint96[] memory multipliers) internal view returns (uint256) {
```

Comment:
dont make this another internal function?

Reply:
isnt it nice to separate out this logic? Feels cluttered to me otherwise

Reply:
ok up to u